### ISSUE TITLE
configuration file for the game does not work

### ISSUE THREAD

**Issue by ricc3** [flightlessmango/MangoHud#1386]
Hello ;) 

**List relevant hardware/software information:**

- Pop!_OS 22.04 LTS
- Steam 1.0.0.79 flatpak
- MangoHud 0.7.2 flatpak

How to use a custom configuration file for a game ?
Take the Gray Zone Warfare game here, for example.
In Steam in the properties of the game I just added:
`PROTON_LOG=1 MANGOHUD_LOG_LEVEL=debug MANGOHUD=1 %command%`

In the generated log I have:
```
Command: ['/media/Game_Disk/SteamLibrary/steamapps/common/Gray Zone Warfare/GZWClientEAC.exe']
...
[2024-07-22 17:06:13.904] [MANGOHUD] [info] [blacklist.cpp:71] process 'explorer.exe' is blacklisted in MangoHud
...
[2024-07-22 17:06:14.872] [MANGOHUD] [debug] [config.cpp:117] skipping config: '/home/ricc3/.config/MangoHud/MangoHud.conf' [ not found ]
[2024-07-22 17:06:14.872] [MANGOHUD] [debug] [overlay_params.cpp:1032] Failed to read presets file: '/home/ricc3/.var/app/com.valvesoftware.Steam/config/MangoHud/presets.conf'.  Falling back to default presets
[2024-07-22 17:06:14.872] [MANGOHUD] [debug] [config.cpp:117] skipping config: '/home/ricc3/.config/MangoHud/MangoHud.conf' [ not found ]
[2024-07-22 17:06:14.872] [MANGOHUD] [debug] [logging.cpp:237] Logger constructed!
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1067] Ram:32764816
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1068] Cpu:AMD Ryzen 7 5800X3D 8-Core Processor
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1069] Kernel:6.9.3-76060903-generic
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1070] Os:Steam Runtime 3 (sniper)
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1071] Driver:
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [overlay.cpp:1072] CPU Scheduler:schedutil
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:507] hwmon: sensor name: k10temp
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:539] hwmon: using input: /sys/class/hwmon/hwmon2/temp1_input
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:632] hwmon: sensor name: k10temp
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:632] hwmon: sensor name: acpitz
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:632] hwmon: sensor name: gigabyte_wmi
[2024-07-22 17:06:14.902] [MANGOHUD] [debug] [cpu.cpp:632] hwmon: sensor name: amdgpu
...
[2024-07-22 17:06:14.915] [MANGOHUD] [debug] [vulkan.cpp:1859] gpu: AMD Radeon RX 5500 XT (RADV NAVI14)
[2024-07-22 17:06:14.915] [MANGOHUD] [debug] [overlay.cpp:874] drm path check: /sys/class/drm/card1
[2024-07-22 17:06:14.915] [MANGOHUD] [debug] [overlay.cpp:916] Using gpu_metrics of /sys/class/drm/card1/device/gpu_metrics
[2024-07-22 17:06:14.915] [MANGOHUD] [debug] [overlay.cpp:974] Uploading is disabled (permit_upload = 0)
...
[2024-07-22 17:06:14.919] [MANGOHUD] [debug] [vulkan.cpp:1546] Present mode: IMMEDIATE
2744.306:0124:0144:warn:threadname:NtSetInformationThread Thread renamed to L"dxvk-frame"
[2024-07-22 17:06:14.921] [MANGOHUD] [debug] [vulkan.cpp:720] Recreating font image
[2024-07-22 17:06:14.940] [MANGOHUD] [debug] [vulkan.cpp:739] Default font tex size: 1024x2048px
```

and all in all, so much for the Mango phrase.

I understand that my file that runs the game is _GZWClientEAC.exe_. So I created the file: _GZWClientEAC.conf_ as well as _wine-GZWClientEAC.conf_ and rarely does either of them work.
Admittedly, I set Steam to also read configuration files from the default location: _/.config/MangoHud/MangoHud.conf_ as well as _/.var/app/com.valvesoftware.Steam/config/MangoHud/presets.conf_ but neither location works.

I just don't know what I can do next....

**Comment by flightlessmango** [comment]
Short answer: don't use flatpak

So you can't set config options at all or just for this specific game?

**Comment by ricc3** [comment]
From my amateur observations - because I am a newcomer to the Linux world - it seems that MangoHUD does not work beyond the Steam container.... That is, I have to keep all configuration files in /.var/app/com.valvesoftware.Steam/.config/MangoHud.

I still don't understand why I was previously able to use the config file from location ~/.config/MangoHud/MangoHud.conf together with Steam.
I have not been able to reproduce this again.

Oh and it turns out that my game doesn't start from GZWClientEAC.exe but GZWClientSteam-Win64-Shipping.exe - by some miracle during one attempt it showed me that it was looking for the wine-GZWClientSteam-Win64-Shipping.conf config file - likewise no matter if I use mangohud or MANGOHUD=1 it didn't show me that it was looking for that file again, and I've already run the game maybe a hundred times to debug.

I also noticed by the way now that Mango has started showing and searching:
```
/home/ricc3/.var/app/com.valvesoftware.Steam/config/MangoHud/wine-GZWClientEAC.conf
/home/ricc3/.var/app/com.valvesoftware.Steam/.local/share/Steam/compatibilitytools.d/GE-Proton9-10/files/bin/MangoHud.conf
/home/ricc3/.var/app/com.valvesoftware.Steam/config/MangoHud/wine64-preloader.conf
/etc/MangoHud.conf
/home/ricc3/.var/app/com.valvesoftware.Steam/config/MangoHud/MangoHud.conf
/home/ricc3/.var/app/com.valvesoftware.Steam/config/MangoHud/presets.conf
```

I honestly have no idea what happened to make such a difference. The only thing I've started doing is debugging.


topic basically closed...

well unless someone knows how to make MangoHud back to be able to load files from:
/home/ricc3/.config/MangoHud/MangoHud.conf
or fonts from the location:
/usr/share/fonts/truetype/ubuntu/UbuntuMono-RI.ttf


_By the way, if someone can't change the font for Mango, you also have to copy it to /.var/app/com.valvesoftware.Steam/.config/MangoHud_
